### PR TITLE
verifying EnvironmentID before creating infrastructure

### LINKED ISSUE #4463: `registerInfra` API allows registering Infra even with invalid `environmentID`
**What happened**:

A project was created without specifying an environment. Surprisingly, when invoking the registerInfra API with a non-existent environmentID, the call returned a success response. Moreover, execution plane components appear to operate normally despite the absence of an associated environment. Although the infrastructure is not listed on the environments page, it does show up in the list when constructing a chaos experiment.

**What you expected to happen**:

The anticipation was that the API would detect the invalid environmentID and return a validation error message, indicating that the provided environmentID does not correspond to a valid environment.

**Where can this issue be corrected? (optional)**

https://github.com/litmuschaos/litmus/blob/5df01f1c386ab142015c03d998f6c18587e01d20/chaoscenter/graphql/server/pkg/chaos_infrastructure/service.go#L74

### PR DESCRIPTION
**Pull request by Dhanush0369**
## Proposed changes
Fixes #4463
Summarize your changes here to communicate with the maintainers and make sure to put the link of that issue

## Types of changes

What types of changes does your code introduce to Litmus? Put an `x` in the boxes that apply
- [ ] New feature (non-breaking change which adds functionality)
- [ x] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices applies)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
- [ ] I have read the [CONTRIBUTING](https://github.com/litmuschaos/litmus/blob/master/CONTRIBUTING.md) doc
- [ ] I have signed the commit for DCO to be passed.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added necessary documentation (if appropriate)

## Dependency
- Please add the links to the dependent PR need to be merged before this (if any).

## Special notes for your reviewer:
graphql_pic

### CODE REVIEW

**Review by vanshBhatia-A4k9 — APPROVED** [review]
LGTM.
@Saranya-jena can you PTAL.

**Comment by Dhanush0369** [comment]
@Saranya-jena shall I close this pr and make another pr. It seems I have merged using the  wrong method.

**Comment by prithvi1307** [comment]
Sure @Dhanush0369 please open another PR if you deem this wrong

File: chaoscenter/graphql/server/pkg/chaos_infrastructure/service.go [hunk 1]
```diff
@@ -86,6 +86,11 @@ func (in *infraService) RegisterInfra(c context.Context, projectID string, input
 		return nil, errors.New("infra already exists in this project with the same name")
 	}
 
+	// Check if EnvironmentID is valid
+	if _, err := in.envOperator.GetEnvironmentDetails(c, input.EnvironmentID, projectID); err != nil {
+		return nil, errors.New("Invalid EnvironmentID")
```
Comment (SarthakJain26) [inline chaoscenter/graphql/server/pkg/chaos_infrastructure/service.go:91]: @Dhanush0369 have you tested these changes?

Comment (Dhanush0369) [reply chaoscenter/graphql/server/pkg/chaos_infrastructure/service.go:91]: Yes, I have attached the graphql playground pic

**Review by SarthakJain26 — APPROVED** [review]

**Review by namkyu1999 — APPROVED** [review]
LGTM 🚀

**Comment by Saranya-jena** [comment]
@Dhanush0369 is this good to merge?

**Review by Saranya-jena — APPROVED** [review]

**Pull request merged**